### PR TITLE
Ddfform 822 modal til abningstider

### PR DESCRIPTION
**Pull request by kasperbirch1**
### Link to Issue
[DDFFORM-822](https://reload.atlassian.net/browse/DDFFORM-822)

### Description
This pull request introduces new `OpeningHoursSidebar` stories for inspecting layout and animation. Additionally, it refactors the `Dialog` component to support sidebar functionality, enhancing the overall flexibility and usability of the dialog system.

I want to use our dialog component as much as possible for accessibility and reduction of JavaScript. However, there are some limitations related to CSS (so far). But I have arrived at a satisfactory result. I had to ignore some CSS linting rules because they do not include the latest CSS features.

<s>I know that Firefox doesn't support @starting-style for display: none (yet). In this case, the animation won't work. However, I believe there are advantages in using simpler HTML, better accessibility, and less JavaScript, so it's worth going with this solution for now.</s>

<S>https://caniuse.com/mdn-css_at-rules_starting-style</s>

**After updating, my Firefox works correctly.**

#### Test
http://varnish.pr-1528.dpl-cms.dplplat01.dpl.reload.dk/frontpage

[DDFFORM-822]: https://reload.atlassian.net/browse/DDFFORM-822?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by xendk — APPROVED** [review]
:+1:

File: src/stories/Library/disclosure-controllable/DisclosureControllable.tsx [hunk 1]
```diff
@@ -25,13 +25,22 @@ const DisclosureControllable: React.FunctionComponent<
...
+  );
+
   const disclosureId = `disclosure-${id}`;
   return (
     <div className={detailsClassName}>
       <div
         className={summaryClassName}
         onClick={toggleOpen}
-        onKeyDown={toggleOpen}
+        onKeyDown={onKeyDown}
```
Comment (kasperg) [inline src/stories/Library/disclosure-controllable/DisclosureControllable.tsx:43]: I am not sure how this is related to the change at hand.

The mention in the commit comment does not help me much. The change might be good but I would have expected it to be a separate commit with more info.

File: src/stories/Library/dialog/dialog.scss [hunk 2]
```diff
@@ -13,3 +13,58 @@
...
+  to {
+    background-color: rgb(0 0 0 / 25%);
+  }
+}
+
+// This sets the correct sidebar width for accurate visual regression testing.
+.storybook-dialog-sidebar {
+  @include media-query__small {
+    width: 435px;
+  }
```
Comment (kasperg) [inline src/stories/Library/dialog/dialog.scss:69]: :+1: Appreciate this comment.

File: src/stories/Library/opening-hours-sidebar/OpeningHoursSidebarDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+}) => {
+  return (
+    <div className="opening-hours-sidebar-details__content">
+      <dl className="opening-hours-sidebar-details__list">
+        {openingHoursData.map(({ term, description }, i) => (
+          <div key={i} className="opening-hours-sidebar-details__item">
+            <dt className="opening-hours-sidebar-details__term">{term}</dt>
+            <dd className="opening-hours-sidebar-details__description">
+              {description}
+            </dd>
```
Comment (kasperg) [inline src/stories/Library/opening-hours-sidebar/OpeningHoursSidebarDetails.tsx:25]: Finally a little `<dl>` love :smile:

File: src/stories/Library/opening-hours-sidebar/opening-hours-sidebar.scss [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+.opening-hours-sidebar-summary__icon {
+  transition: transform 0.3s;
+}
+
+.opening-hours-sidebar-summary[aria-expanded="true"]
+  .opening-hours-sidebar-summary__icon {
+  transform: rotateZ(180deg);
+}
```
Comment (kasperg) [inline src/stories/Library/opening-hours-sidebar/opening-hours-sidebar.scss:44]: I wonder why this is necessary. I would have expected it to be handled by our disclosure component.

**Review by kasperg — APPROVED** [review]
:+1: LGTM

File: src/stories/Library/disclosure-controllable/DisclosureControllable.tsx (hunk 1 above)
Comment (kasperbirch1) [reply src/stories/Library/disclosure-controllable/DisclosureControllable.tsx:43]: Apologies, I should have explained this more clearly. The change was because I was testing accessibility with the keyboard and noticed it was not working as expected. After reviewing our `dpl-react` implementation, I found that it was set up correctly there. As a result, I decided to add the necessary changes to ensure proper functionality.

File: src/stories/Library/opening-hours-sidebar/opening-hours-sidebar.scss (hunk 4 above)
Comment (kasperbirch1) [reply src/stories/Library/opening-hours-sidebar/opening-hours-sidebar.scss:44]: There is generally some confusion in our CSS regarding these arrows. I suggest we create a separate ticket to standardize the logic in both CMS and React.

Comment (kasperbirch1) [reply src/stories/Library/opening-hours-sidebar/opening-hours-sidebar.scss:49]: https://reload.atlassian.net/browse/DDFHER-75
